src/app/shared/log-in/methods/orcid/log-in-orcid.component.ts

```diff
@@ -0,0 +1,110 @@
+import {CoreState} from "../../../../core/core-state.model";
+
+@Component({
+  selector: 'ds-log-in-orcid',
+  templateUrl: './log-in-orcid.component.html',
+})
+@renderAuthMethodFor(AuthMethodType.Orcid)
+export class LogInOrcidComponent implements OnInit {
+
+  /**
+   * The authentication method data.
+   * @type {AuthMethod}
+   */
+  public authMethod: AuthMethod;
+
+  /**
+   * True if the authentication is loading.
+   * @type {boolean}
+   */
+  public loading: Observable<boolean>;
+
+  /**
+   * The orcid authentication location url.
+   * @type {string}
+   */
+  public location: string;
+
+  /**
+   * Whether user is authenticated.
+   * @type {Observable<string>}
+   */
+  public isAuthenticated: Observable<boolean>;
+
+  /**
+   * @constructor
+   * @param {AuthMethod} injectedAuthMethodModel
+   * @param {boolean} isStandalonePage
+   * @param {NativeWindowRef} _window
+   * @param {AuthService} authService
+   * @param {HardRedirectService} hardRedirectService
+   * @param {Store<State>} store
+   */
+  constructor(
+    @Inject('authMethodProvider') public injectedAuthMethodModel: AuthMethod,
+    @Inject('isStandalonePage') public isStandalonePage: boolean,
+    @Inject(NativeWindowService) protected _window: NativeWindowRef,
+    private authService: AuthService,
+    private hardRedirectService: HardRedirectService,
+    private store: Store<CoreState>
+  ) {
+    this.authMethod = injectedAuthMethodModel;
+  }
+
+  ngOnInit(): void {
+    // set isAuthenticated
+    this.isAuthenticated = this.store.pipe(select(isAuthenticated));
+
+    // set loading
+    this.loading = this.store.pipe(select(isAuthenticationLoading));
+
+    // set location
+    this.location = decodeURIComponent(this.injectedAuthMethodModel.location);
+
+  }
+
+  redirectToOrcid() {
+
+    this.authService.getRedirectUrl().pipe(take(1)).subscribe((redirectRoute) => {
+      if (!this.isStandalonePage) {
+        redirectRoute = this.hardRedirectService.getCurrentRoute();
+      } else if (isEmpty(redirectRoute)) {
+        redirectRoute = '/';
+      }
+      const correctRedirectUrl = new URLCombiner(this._window.nativeWindow.origin, redirectRoute).toString();
+
+      let orcidServerUrl = this.location;
+      const myRegexp = /\?redirectUrl=(.*)/g;
+      const match = myRegexp.exec(this.location);
+      const redirectUrlFromServer = (match && match[1]) ? match[1] : null;
+
+      // Check whether the current page is different from the redirect url received from rest
+      if (isNotNull(redirectUrlFromServer) && redirectUrlFromServer !== correctRedirectUrl) {
+        // change the redirect url with the current page url
+        const newRedirectUrl = `?redirectUrl=${correctRedirectUrl}`;
+        orcidServerUrl = this.location.replace(/\?redirectUrl=(.*)/g, newRedirectUrl);
+      }
+
+      // redirect to orcid authentication url
+      this.hardRedirectService.redirect(orcidServerUrl);
+    });
+
+  }
```

Comment:
This is an almost exact copy of [`redirectToShibolleth()`](https://github.com/4Science/dspace-angular/blob/716cea376da914d56f1f96a7aab3d7bc69a63f07/src/app/shared/log-in/methods/shibboleth/log-in-shibboleth.component.ts#L87-L113) in the `LogInShibbolethComponent`, only a few variable names appear to have changed. Perhaps they can be refactored to use the same method?